pkg/agent/loadbalancer/servers.go

```diff
@@ -21,28 +21,37 @@ import (
 
 var defaultDialer proxy.Dialer = &net.Dialer{}
 
-func init() {
+// SetHTTPProxy configures a proxy-enabled dialer to be used for all loadbalancer connections,
+// if the agent has been configured to allow use of a HTTP proxy, and the environment has been configured
+// to indicate use of a HTTP proxy for the server URL.
+func SetHTTPProxy(address string) error {
 	// Check if env variable for proxy is set
-	address := os.Getenv(version.ProgramUpper + "_URL")
-
-	if useProxy, _ := strconv.ParseBool(os.Getenv(version.ProgramUpper + "_AGENT_HTTP_PROXY_ALLOWED")); !useProxy {
-		return
+	if useProxy, _ := strconv.ParseBool(os.Getenv(version.ProgramUpper + "_AGENT_HTTP_PROXY_ALLOWED")); !useProxy || address == "" {
```

Comment:
are we good not handling err or logging if i set a wrong value for boolean?

Reply:
anything that's not parsed as a true bool (`1, t, T, TRUE, true, True`) is just used as false, that's fine.